lib/src/currency_cloud_base.dart

```diff
@@ -11,48 +14,81 @@ abstract class CurrencyCloudApi {
   CurrencyCloudApi.withCurrencyCloudClient(this._authToken, this.client);
 }
 
+/// [CurrencyCloudClient] is used for communication with the CurrencyCloudService. It provides Request methods
+/// like [get] and [post] which handle the basic communication overheads like adding authentication headers.
 class CurrencyCloudClient {
-  final String baseUrl = 'https://devapi.thecurrencycloud.com/v2';
+  final String baseUri = 'https://devapi.thecurrencycloud.com/v2';
   AuthToken _authToken;
 
   CurrencyCloudClient(this._authToken);
 
   /// Sets auth headers in provided [headers] and sends HTTP GET request to
-  /// given methodUrl. Beware [headers] are being modified!
-  get(String methodUrl, {Map<String, String> headers}) async {
-    final String url = baseUrl + methodUrl;
+  /// given [uri] with [body] set as encoded uri parameters.
+  Future<Map<String, String>> get(String uri, {Map<String, String> body, Map<String, String> headers}) async {
+    uri = baseUri + uri;
+
+    headers = _setAuthHeader(headers);
+
+    if (body != null) {
+      // list of the uri parameters to be set
+      var params = [];
+      for (var key in body.keys) {
+        params.add('$key=${body[key]}');
+      }
+
+      uri += '?';
+      uri += params.join('&');
+      uri = Uri.encodeFull(uri);
```

Comment:
That's a bit dangerous. I would collect the different segments (`scheme`, `host`, `queryParameters`, etc...) and build the `Uri` in one go here: `new Uri(/* params */);`